core/templates/cowdata.h

```diff
@@ -232,7 +232,7 @@ uint32_t CowData<T>::_copy_on_write() {
 
 		uint32_t *mem_new = (uint32_t *)Memory::alloc_static(_get_alloc_size(current_size), true);
 
-		new (mem_new - 2, sizeof(uint32_t), "") SafeNumeric<uint32_t>(1); //refcount
+		new (mem_new - 2) SafeNumeric<uint32_t>(1); //refcount
```

Comment:
This produces warnings on G++ 11.1.0, the current version on Arch.

```
./core/templates/cowdata.h:235:30: warning: placement new constructing an object of type 'SafeNumeric<unsigned int>' and size '4' in a region of type 'uint32_t*' {aka 'unsigned int*'} and size '0' [-Wplacement-new=]
  235 |                 new (mem_new - 2) SafeNumeric<uint32_t>(1); //refcount
      |                      ~~~~~~~~^~~
./core/templates/cowdata.h:233:27: note: at offset -8 from 'mem_new' declared here
  233 |                 uint32_t *mem_new = (uint32_t *)Memory::alloc_static(_get_alloc_size(current_size), true);
      |                           ^~~~~~~
```

Reply:
This warning was reported here already: https://github.com/godotengine/godot/issues/52119